pkg/controller/cluster/main-controller.go

```diff
@@ -1507,12 +1507,17 @@ func (c *Controller) processDelete(ctx context.Context, tenant *miniov1.Tenant)
 	if err != nil {
 		return err
 	}
-	// Remove finalizer
-	for i, f := range tenant.ObjectMeta.Finalizers {
-		if f == miniov1.Finalizer {
-			tenant.ObjectMeta.Finalizers = append(tenant.Finalizers[:i], tenant.ObjectMeta.Finalizers[i+1:]...)
+
+	// Remove miniov1 finalizers
+	n := 0
+	for _, f := range tenant.ObjectMeta.Finalizers {
+		if f != miniov1.Finalizer {
+			tenant.ObjectMeta.Finalizers[n] = f
+			n++
 		}
 	}
+	tenant.ObjectMeta.Finalizers = tenant.ObjectMeta.Finalizers[:n]
+
```

Comment:
More idiomatic way to do this would be 

```go
finalizers := tenant.ObjectMeta.Finalizers[:0]
for _, f := range tenant.ObjectMeta.Finalizers {
         if f != miniov1.Finalizer {
                   finalizers = append(finalizers, f)
         }
}
tenant.ObjectMeta.Finalizers = finalizers
```

https://play.golang.org/p/Cr-4HXv4Oj5